core/Tracker/Visit.php

```diff
@@ -166,6 +166,12 @@ public function handle()
         // AND
         // - the last page view for this visitor was less than 30 minutes ago @see isLastActionInTheSameVisit()
         if (!$isNewVisit) {
+            // on a ping request that is received before the standard visit length, we just update the visit time w/o
+            // adding a new action
+            if ($this->isPingRequest()) {
+                $action = null;
+            }
+
```

Comment:
I think it would be nice to ignore all "goal" related stuff for a ping request as it could improve performance quite a bit but it might increase code complexity a lot so probably it's not worth it until it becomes a problem. 

I had a look through all the defined dimensions and to possible other side effects due to re-sending the last request in `getRequest('ping=1')` in `piwik.js` but looks good. It's really hard to tell though, I reckon we will see. We just need to make sure not to create duplicated visitors eg for intranet etc. and it looks OK. 

just FYI :)